### PR TITLE
[#2] simple emp service

### PR DESCRIPTION
**Pull request by sumindaaa**
Please review this

### CODE REVIEW

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.sumindaa.demo.employeeservice.model;
+
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java:2]: Need all properties mentioned in  #2

File: src/main/java/com/sumindaa/demo/employeeservice/controller/Hello2.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+class Hello2 {
+    @RequestMapping("/hello2")
+    public String sayHi(){
+        return "hiii";
+    }
+
+}
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/controller/Hello2.java:13]: Do not commit unwanted files

File: src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.sumindaa.demo.employeeservice.controller;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+class EmployeeController {
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java:11]: as mentioned in #2 we should have REST endpoints for all CRUD operations
CRUD, Create, Read, Update, Delete

File: pom.xml [hunk 4]
```diff
@@ -5,13 +5,13 @@
 	<parent>
```
Comment (Chamindasl) [inline pom.xml:5]: Before comit your change, make sure to format all of your files nicely

**Review by Chamindasl — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 5]
```diff
@@ -33,6 +33,11 @@
 			<artifactId>h2</artifactId>
 			<scope>runtime</scope>
 		</dependency>
+		<dependency>
+			<groupId>mysql</groupId>
+			<artifactId>mysql-connector-java</artifactId>
+			<scope>runtime</scope>
```
Comment (Chamindasl) [inline pom.xml]: why you added this dependancy. you no need this now.

File: pom.xml [hunk 6]
```diff
@@ -5,13 +5,13 @@
 	<parent>
 		<groupId>org.springframework.boot</groupId>
```
Comment (Chamindasl) [inline pom.xml:6]: I think you are using tabs for indentation.  use spaces instead for tabs. you no need to change manually. Intellij can do that when you press tab.

**Review by Chamindasl — CHANGES_REQUESTED** [review]
Please make changes requested

File: pom.xml [hunk 7]
```diff
@@ -5,13 +5,13 @@
...
 		<artifactId>spring-boot-starter-parent</artifactId>
-		<version>2.2.0.RELEASE</version>
+		<version>2.2.4.RELEASE</version>
 		<relativePath/> <!-- lookup parent from repository -->
 	</parent>
-	<groupId>sumindaaa</groupId>
-	<artifactId>smart-employee-service</artifactId>
+	<groupId>com.sumindaa.demo</groupId>
+	<artifactId>employee-service</artifactId>
```
Comment (Chamindasl) [inline pom.xml]: I think smart-employee-service is the correct artifactid

**Review by Chamindasl — CHANGES_REQUESTED** [review]
Request changes

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java (hunk 1 above)
Comment (sumindaaa) [reply src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java:2]: done

File: src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java (hunk 3 above)
Comment (sumindaaa) [reply src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java:11]: Done. However still I didn't do the service method and will do soon

File: pom.xml (hunk 5 above)
Comment (sumindaaa) [reply pom.xml]: then how we connect a database

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java [hunk 8]
```diff
@@ -1,5 +1,11 @@
 package com.sumindaa.demo.employeeservice.model;
 
-public class Employee {
+import java.util.Date;
 
+public class Employee {
+int id;
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java]: Make these private and create getter setters

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java [hunk 9]
```diff
@@ -1,5 +1,11 @@
...
+char firstName;
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java]: Again wrong format

File: pom.xml (hunk 5 above)
Comment (Chamindasl) [reply pom.xml]: Yes, but for the now we dont have db right.

**Review by Chamindasl — CHANGES_REQUESTED** [review]

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java (hunk 9 above)
Comment (sumindaaa) [reply src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java]: done

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java (hunk 8 above)
Comment (sumindaaa) [reply src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java]: done

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java [hunk 10]
```diff
@@ -3,9 +3,49 @@
 import java.util.Date;
 
 public class Employee {
-int id;
-char firstName;
-char lastName;
-Date dob;
-char gender;
+    private int id;
+    private char firstName;
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java]: Why they are char, char is only for a single character, you should use string.

File: src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java [hunk 11]
```diff
@@ -10,4 +11,22 @@ public String getAllTopics(){
         return "all topics";
 
     }
+    @RequestMapping ("/employees/{id}")
+    public Employee getEmployee(@PathVariable int id){
+        return EmployeeService.getEmployee(id);
+    }
+@RequestMapping (method = RequestMethod.POST, value = "/employees")
+    public void addEmployee(@RequestBody Employee employee){
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java]: Could you able to access EmployeeService this way? 
EmployeeService.getEmployee(id);

Is this compiling?

**Review by Chamindasl — CHANGES_REQUESTED** [review]

File: src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java (hunk 11 above)
Comment (sumindaaa) [reply src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java]: Done. I made them static

File: src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java (hunk 10 above)
Comment (sumindaaa) [reply src/main/java/com/sumindaa/demo/employeeservice/model/Employee.java]: Done

File: src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java (hunk 11 above)
Comment (Chamindasl) [reply src/main/java/com/sumindaa/demo/employeeservice/controller/EmployeeController.java]: Can you do a mvn clean install and confirm everything is OK, 
AND this is very important, before every push to origin. You can understand why?

File: pom.xml (hunk 7 above)
Comment (Chamindasl) [reply pom.xml]: Can yo correct this?

File: pom.xml (hunk 5 above)
Comment (Chamindasl) [reply pom.xml]: Can you correct this?

File: src/main/java/com/sumindaa/demo/employeeservice/controller/Hello2.java (hunk 2 above)
Comment (Chamindasl) [reply src/main/java/com/sumindaa/demo/employeeservice/controller/Hello2.java:13]: Correct this

File: src/main/java/com/sumindaa/demo/employeeservice/service/EmployeeService.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
+package com.sumindaa.demo.employeeservice.service;
```
Comment (Chamindasl) [inline src/main/java/com/sumindaa/demo/employeeservice/service/EmployeeService.java:1]: I need the implementation of this. You no need to use a database. You can use Simple Map, Set or List whatever you think suits best.

**Review by Chamindasl — CHANGES_REQUESTED** [review]

File: src/main/java/com/sumindaa/demo/employeeservice/controller/Hello2.java (hunk 2 above)
Comment (sumindaaa) [reply src/main/java/com/sumindaa/demo/employeeservice/controller/Hello2.java:13]: I think I did it now. however I'm not sure. when I try to commit it makes an issue.

File: pom.xml (hunk 5 above)
Comment (sumindaaa) [reply pom.xml]: done

File: pom.xml (hunk 7 above)
Comment (sumindaaa) [reply pom.xml]: I think I done it correctly now

**Comment by Chamindasl** [comment]
I am merging your full request. Its not prefect, but not bad as a beginner. 
See bellow
https://github.com/sumindaaa/smart-employee-service/network

Before 
image

After
image

**Pull request merged**